CombatCrittersSharp/objects/card/CardQuery.cs

```diff
@@ -1,18 +1,68 @@
 using CombatCrittersSharp.objects.card.Interfaces;
+using static CombatCrittersSharp.objects.card.Interfaces.ICardQuery;
 
 namespace CombatCrittersSharp.objects.card
 {
     public class CardQuery : ICardQuery
     {
-        private readonly string _queryString;
-        public CardQuery(string queryString)
+        private readonly int _cost;
+        private readonly bool _costLess;
+        private readonly List<int> _ids;
+        private readonly CardOrder _order;
+        private readonly bool  _owned;
+        private readonly List<int> _rarities;
+        private readonly bool _raritiesInclude;
+
+        public CardQuery(int cost, bool costLess, List<int> ids, CardOrder order, bool owned, List<int> rarities, bool rarityInclude)
         {
-            _queryString = queryString;
+            _cost = cost;
+            _costLess = costLess;
+            _ids = ids;
+            _order = order;
+            _owned = owned;
+            _rarities = rarities;
+            _raritiesInclude = rarityInclude;
         }
 
+        /// <summary>
+        /// This maps each CardOder enum value to its corresponding string.
+        /// </summary>
+        /// <param name="order"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException"></exception>
+        private static string MapOrderToString(CardOrder order)
+        {
+            return order switch
+            {
+                CardOrder.NONE => "",
+                CardOrder.ID => "ID",
+                CardOrder.NAME => "NAME",
+                CardOrder.PLAYCOST => "PLAY_COST",
+                CardOrder.RARITY => "RARITY",
+                _ => throw new ArgumentOutOfRangeException(nameof(order), order, null)
```

Comment:
is there no way to define the enum with the strings?

Reply:
No, only if I use the enum ToString method. However, mapping works as well